src/components/connect.js

```diff
@@ -17,9 +17,36 @@ function getDisplayName(WrappedComponent) {
 // Helps track hot reloading.
 let nextVersion = 0
 
-export default function connect(mapPropsToRequestsToProps, options = {}) {
+function connectFactory(options = {}) {
+  function connectImpl(map) {
+    return connect(map, options)
+  }
+
+  connectImpl.withFetch = function (fetch) {
+    invariant(
+      typeof fetch === 'function',
+      '`withFetch` must be passed a function. Instead received a %s.',
+      typeof fetch
+    )
+
+    return connectFactory(assign({}, options, { fetch: fetch }))
+  }
+
+  connectImpl.withRef = function (withRef = true) {
+    return connectFactory(assign({}, options, { withRef: withRef }))
+  }
+
+  return connectImpl
+}
+
+export default connectFactory()
+
+function connect(mapPropsToRequestsToProps, options) {
   const finalMapPropsToRequestsToProps = mapPropsToRequestsToProps || defaultMapPropsToRequestsToProps
-  const { withRef = false } = options
+  const {
+    fetch = window.fetch,
+    withRef = false
```

Comment:
I like that we are dropping the options hash here, but I'm slightly worried that that is a breaking change. What do you think about still allowing it in the options hash with something that emits a warning? We can leave it in there for one release and then pull it out.